presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MetadataDeleteOptimizer.java

```diff
@@ -89,10 +91,20 @@ public PlanNode visitTableFinish(TableFinishNode node, RewriteContext<Void> cont
                 return context.defaultRewrite(node);
             }
             TableScanNode tableScanNode = tableScan.get();
-            if (!metadata.supportsMetadataDelete(session, tableScanNode.getTable(), tableScanNode.getLayout().get())) {
+            if (!metadata.supportsMetadataDelete(session, tableScanNode.getTable())) {
                 return context.defaultRewrite(node);
             }
-            return new MetadataDeleteNode(idAllocator.getNextId(), delete.get().getTarget(), Iterables.getOnlyElement(node.getOutputSymbols()), tableScanNode.getLayout().get());
+            DeleteHandle currentDeleteHandle = delete.get().getTarget();
+            return new MetadataDeleteNode(
+                    idAllocator.getNextId(),
+                    new DeleteHandle(
```

Comment:
Curious: Why cannot we do something similar to https://github.com/prestosql/presto/pull/363/commits/47ce7e62491691eb91126c68db09f5eca1e080ff#diff-bceac05c41f5490bdb011d6f1fb7c664R96